cli/azd/cmd/container.go

```diff
@@ -214,6 +215,35 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 			return env, nil
 		},
 	)
+	container.RegisterSingleton(func(
+		ctx context.Context,
+		console input.Console,
+		accountManager account.Manager,
+		userProfileService *azcli.UserProfileService,
+		subResolver account.SubscriptionTenantResolver,
+	) environment.EnvironmentResolver {
+		return func() (*environment.Environment, error) {
+			azdCtx, err := azdcontext.NewAzdContext()
```

Comment:
Instead of loading it here, I think this should consume the `azdContext` available in the container

Reply:
This avoids walking up the directory tree multiple times.

Reply:
Can't use the `azdContext` from the container b/c the app returns an error when trying to create the `azdContext` if you are running azd outside a project.  I need to ignore that issue and not fail.

If you are worried about the tree multiple times, we could change the azdContext within the container to use a resolver. But I don't think it's worth it